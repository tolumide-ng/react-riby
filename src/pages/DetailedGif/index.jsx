import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchDetailedGifAction } from '../../store/modules/detailedGif/actions';
import Loader from '../../main/components/Loader';

const DetailedGif = ({
  location,
  fetchDetailedGif,
  detailedGif = {},
  detailedGifStatus,
  history,
}) => {
  useEffect(() => {
    fetchDetailedGif({ history, id: location.pathname });
  }, []);
  return (
    <div className="w-full flex-col justify-center items-center mt-10">
      <div className="flex items-center justify-center flex-col">
        <Link to="/" className="mb-4">
          Go Home
        </Link>
        {detailedGifStatus === 'pending' && (
          <div data-testid="pageloading" className="mt-20">
            <Loader />
          </div>
        )}
      </div>

      {detailedGifStatus === 'success' && (
        <div className="w-full flex items-center justify-center flex-col">
          <img
            src={
              detailedGif &&
              detailedGif.images &&
              detailedGif.images.original &&
              detailedGif.images.original.url
            }
            alt={detailedGif && detailedGif.title}
            className="object-cover"
            style={{ height: '600px' }}
          />
          <div className="w-9/12 mt-2">
            <div className="flex">
              <strong className="mr-2">Title: </strong>
              <p>{detailedGif && detailedGif.title}</p>
            </div>
            <div className="flex">
              <strong className="mr-2">Rating: </strong>
              <p>{detailedGif && detailedGif.rating}</p>
            </div>
            <div className="flex">
              <strong className="mr-2">Source: </strong>
              <p>{detailedGif && detailedGif.source}</p>
            </div>
            <div className="flex">
              <strong className="mr-2">Username: </strong>
              <p>{detailedGif && detailedGif.username}</p>
            </div>
          </div>
        </div>
      )}
    </div>
  );
};

const mapStateToProps = state => ({
  detailedGif: state.detailedGifReducer.detailedGif,
  detailedGifError: state.detailedGifReducer.detailedGifError,
  detailedGifStatus: state.detailedGifReducer.detailedGifStatus,
});

const mapDispatchToProps = dispatch => ({
  fetchDetailedGif: ({ history, id }) =>
    dispatch(fetchDetailedGifAction({ history, id })),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(DetailedGif);
