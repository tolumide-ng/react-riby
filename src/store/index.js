import { createStore, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import rootReducer from './modules';

const middlewares = applyMiddleware(ReduxThunk);

const store = createStore(rootReducer, composeWithDevTools);

export default store;
