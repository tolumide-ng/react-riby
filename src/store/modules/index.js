import { combineReducers } from 'redux';
import allGifReducer from './allGifs/reducers';
import detailedGifReducer from './detailedGif/reducers';
import searchGifReducer from './searchGif/reducers';

const rootReducer = combineReducers({
  allGifReducer,
  detailedGifReducer,
  searchGifReducer,
});

export default rootReducer;
