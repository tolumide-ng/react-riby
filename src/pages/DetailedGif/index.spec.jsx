/* eslint-disable react/jsx-props-no-spreading */
import { render, fireEvent, cleanup } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import DetailedGif from './index';
import store from '../../store';

const history = { push: jest.fn() };
const location = { pathname: '/pathname' };
const fetchDetailedGif = jest.fn(() => {
  return [
    {
      id: '45',
      title: 'title',
      source: 'source',
      rating: 'rating',
      username: 'username',
      images: {
        original: {
          url: 'anotherone',
        },
      },
    },
  ];
});

const detailedGifStatus = '';

describe('DetailedGif Page', () => {
  test('Should render the Detailed Gif Page', () => {
    const defaultProps = {
      history,
      location,
      fetchDetailedGif,
      detailedGifStatus,
    };
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <DetailedGif {...defaultProps} />
        </MemoryRouter>
      </Provider>,
    );
    expect(asFragment).toMatchSnapshot();
  });

  test('DetailedGif Page should render loading if detailed status is pending', () => {
    const defaultProps = {
      history,
      location,
      fetchDetailedGif,
      detailedGifStatus: 'pending',
    };
    const { asFragment, getByTestId } = render(
      <Provider store={store}>
        <MemoryRouter>
          <DetailedGif {...defaultProps} />
        </MemoryRouter>
      </Provider>,
    );
    expect(getByTestId('pageloading')).toBeTruthy();
    expect(asFragment).toMatchSnapshot();
  });
});
