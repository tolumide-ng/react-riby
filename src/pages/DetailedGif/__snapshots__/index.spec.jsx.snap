// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedGif Page DetailedGif Page should render loading if detailed status is pending 1`] = `[Function]`;

exports[`DetailedGif Page Should render the Detailed Gif Page 1`] = `[Function]`;
